modules/utils.lua

```diff
@@ -142,6 +142,26 @@ function GenerateRandomPlate(pattern) -- luacheck: ignore
     return newPattern:upper()
 end
 
+--- maps a table by the given subfield
+--- @param subfield string
+--- @param table table
+--- @return table
+function MapTableBySubfield(subfield, table)
+	local mappedTable = {}
+	for tableIndex, tableData in pairs(table) do
+		if not tableData[subfield] then
```

Comment:
store tableData[subfield] in a variable. It's used several times and will increase readability to extract it.